clients/venice-samza/src/main/java/com/linkedin/venice/samza/VeniceSystemProducer.java

```diff
@@ -257,17 +349,28 @@ public synchronized void start() {
     }
     this.isStarted = true;
 
-    this.d2Client = new D2ClientBuilder().setZkHosts(veniceD2ZKHost)
+    this.primaryControllerColoD2Client = new D2ClientBuilder().setZkHosts(primaryControllerColoD2ZKHost)
+        .setSSLContext(sslFactory.map(SSLFactory::getSSLContext).orElse(null))
+        .setIsSSLEnabled(sslFactory.isPresent())
+        .setSSLParameters(sslFactory.map(SSLFactory::getSSLParameters).orElse(null))
+        .setFsBasePath(primaryControllerColoD2FsBasePath)
+        .setEnableSaveUriDataOnDisk(true)
+        .build();
+
+    this.childColoD2Client = new D2ClientBuilder().setZkHosts(veniceChildD2ZkHost)
```

Comment:
If 'veniceChildD2ZkHost' is the same as 'primaryControllerColoD2ZKHost', can we just refer to the same D2 client instance created above? Will it cause any issue?

Reply:
Also discussed offline -
I prefer keeping separate objects to reduce code complexity. Else, we'll have to keep extra if else checks in multiple places

Tried checking if calling `D2ClientUtils.shutdownClient` on a `D2Client` object is safe - but I saw errors.

Leaving it as two separate objects